tests/unit/lifecycle/LifecycleConductor.spec.js

```diff
@@ -48,6 +48,32 @@ describe('Lifecycle Conductor', () => {
     describe('processBuckets', function test() {
         // timeout set to 4000 to account for 2s for async ops + 1s for bucket queue completion check interval
         this.timeout(4000);
+
+        it('should call callback with error when mongodbclient returns an error', done => {
+            conductor._mongodbClient = { getIndexingJobs: () => {}, getCollection: () => (
+                { find: () => (
+                    { project: () => (
+                        { hasNext: () => new Promise((resolve, reject) => {
+                            reject(new Error('error'));
+                        })
+                    })
+                })
+            }) };
+            conductor._zkClient = { getData: () => {} };
+
+            sinon.stub(conductor, '_controlBacklog')
+                .callsFake(cb => cb(null));
+            sinon.stub(conductor._mongodbClient, 'getIndexingJobs')
+                .callsFake((_, cb) => cb(null, ['job1', 'job2']));
```

Comment:
this stub is redundant with defniition on line 53 (`getIndexingJobs: () => {}`), should keep only one of the two...

more generally, looking at the kind of stubs you are defining here: either use sinon stubs or create non-configurable callbacks manually, but avoid mixing the two approaches...

Reply:
In that case I can't use sinon.stub because I can't stub non-existent property..